### ISSUE TITLE
P17 Gen 2i: Failed ME firmware. No sound on Ubuntu

### ISSUE THREAD

**Issue by RayyanRiaz** [fwupd/firmware-lenovo#280]
The firmware update from 1.22 to 1.23 was unsuccessful, and now at every reboot/start, I get into a screen which says:

`Reading ME Firmware...Please do not power off! 30% Complete`
After 48% progress, it restarts and quickly goes to 10% and then jumps to 60%, and then fails, saying:
` Failed to recover ME Firmware...`

Also, this thing happens twice before I can go to BIOS/grub.
I am giving these details because the issue seems similar to [226](https://github.com/fwupd/firmware-lenovo/issues/226).

Using `sudo dmidecode | grep SKU` I verified that I have a corporate SKU.

### 1: Automatic Ubuntu update of firmware failed: 
Normally I do not have to install these updates, as they are automatically installed by ubuntu software & updates manager. I only noticed this issue when the automatic update from 1.22 to 1.23 failed. 

### 2: I tried to reinstall version 1.23 manually: 
Since I have ubuntu only, I used this link: https://download.lenovo.com/pccbbs/mobiles/n37ul10w.zip
The relevant file was N37ET42W.cab, because N37HT27W.cab is the same as in 1.22.
I used this command to force the reinstall: `fwupdmgr --allow-reinstall --force install N37ET42W.cab`
This did not solve the issue.

### 3: I tried to downgrade to version 1.22 manually: 
Now I tried to downgrade by downloading the old version from this link: https://download.lenovo.com/pccbbs/mobiles/n37ul09w.zip
I used this command: `fwupdmgr --allow-older --allow-reinstall --force install N37ET41W.cab`
This also did not solve the issue.
 
### 4: Since the error was with ME, I also tried manually installing that:
First, I tried installing the latest version(https://download.lenovo.com/pccbbs/mobiles/n37rk02w.zip) and then downgrading it to previous version (https://download.lenovo.com/pccbbs/mobiles/n37rk01w.zip). 
I used the command as mentioned in the readme i.e. `fwupdmgr --allow-older --allow-reinstall install <cab_filename.cab>`
But that also did not solve the problem. Currently the ME version in BIOS seems to be the latest one (15.0.41.2142) ~~, which seems to indicate that the update has been partially installed. Otherwise the version should have been 15.0.35.1951~~ which is correct because I updated the firmware last time around end of Sept. 2022.

### 5: Tried using bootable usb (https://download.lenovo.com/pccbbs/mobiles/n37ur10w.iso):
This did not work because I did not find a way to downgrade using a bootable usb. 

## Output of `fwupdmgr update --show-all`

```

Devices with no available firmware updates: 
 • Integrated Camera
 • SSD 970 EVO Plus 1TB
 • ThinkPad Thunderbolt 4 Dock
 • UEFI Device Firmware
 • UEFI Device Firmware
 • UEFI Device Firmware
 • UEFI Device Firmware
 • UEFI Device Firmware
 • UEFI Device Firmware
 • UEFI Device Firmware
Devices with the latest available firmware version:
 • Intel Management Engine
 • Micron MTFDKBA1T0TFH
 • Prometheus
 • Prometheus IOTA Config
╔══════════════════════════════════════════════════════════════════════════════╗
║ Upgrade System Firmware from 0.1.22 to 0.1.23?                               ║
╠══════════════════════════════════════════════════════════════════════════════╣
║ Lenovo ThinkPad P15 Gen 2i/ P17 Gen 2i/ P15g Gen 2i System Firmware Version  ║
║ 1.23                                                                         ║
║                                                                              ║
║ Important updates:- Update includes a security fix.                          ║
║                                                                              ║
║ New functions or enhancements:- Supported a new option to enable/disable     ║
║ Fingerprint Single-Touch Authentication in BIOS Setup.- Updated the          ║
║ iagnostics module to version 04.26.000.- Supported 5.15-5.35 GHz band in     ║
║ Indonesia.                                                                   ║
║                                                                              ║
║ Problem fixes:- Fixed an issue where no display on the external monitor via  ║
║ docking station while Intel Management Engine Firmware is updated.           ║
║                                                                              ║
║ • Fixed an issue where ethernet connection of the Lenovo USB-C to            ║
║ Ethernet Adapter might be disconnected sometimes.                            ║
║ • Fixed an issue where plugging and unplugging the docking station can       ║
║ cause a Yellow Bang for USB 2.0 devices and can also cause a system hang at  ║
║ reboot.                                                                      ║
║ • Fixed an issue where EC version display as 0.0.0 when EC was updated       ║
║ via LVFS under lower power(25%).                                             ║
║                                                                              ║
║ Do not turn off your computer or remove the AC adapter while the update is   ║
║ in progress.                                                                 ║
║                                                                              ║
║ 20YU004VGE must remain plugged into a power source for the duration of the   ║
║ update to avoid damage.                                                      ║
╚══════════════════════════════════════════════════════════════════════════════╝

```

## Output of `fwupdmgr get-history`

```
20YU004VGE
│
├─Micron MTFDKBA1T0TFH:
│ │   Device ID:          04e17fcf7d3de91da49a163ffe4907855c3648be
│ │   Previous version:   0003P7LN
│ │   Update State:       Success
│ │   Last modified:      2022-10-20 11:03
│ │   GUID:               aebc2e6e-4b99-5cb8-b72f-01039bcb849c
│ │   Device Flags:       • Internal device
│ │                       • Updatable
│ │                       • System requires external power source
│ │                       • Supported on remote server
│ │                       • Needs a reboot after installation
│ │                       • Device is usable for the duration of the update
│ │ 
│ └─3400:
│       New version:      0004P7LN
│       Remote ID:        lvfs
│       Release ID:       12648
│       Summary:          Micron 3400 NVMe SSD Firmware for Lenovo PC
│       License:          Proprietary
│       Size:             1,6 MB
│       Created:          2022-03-22
│       Urgency:          High
│       Vendor:           Lenovo
│       Description:      
│       Supported devices and firmware version: Micron MTFDKBA***TFH 0003P7LN
│     
├─Prometheus:
│ │   Device ID:          0d5d05911800242bb1f35287012cdcbd9b381148
│ │   Previous version:   10.01.3273255
│ │   Update State:       Success
│ │   Last modified:      2022-10-20 11:03
│ │   GUID:               09e8bf16-3e69-50f5-bb66-c7a040248352
│ │   Device Flags:       • Updatable
│ │                       • Supported on remote server
│ │                       • Is in bootloader mode
│ │                       • Cryptographic hash verification is available
│ │ 
│ └─Prometheus Fingerprint Reader:
│       New version:      10.01.3478575
│       Remote ID:        lvfs
│       Release ID:       10694
│       Summary:          Firmware for the Synaptics Prometheus Fingerprint Reader device
│       License:          Proprietary
│       Size:             425,5 kB
│       Created:          2021-06-09
│       Urgency:          Medium
│       Vendor:           Synaptics
│       Duration:         2 seconds
│       Description:      
│       New features and enhancements:
│       
│       • Fix an authentication issue on FOP project.
│     
├─UEFI Device Firmware:
│ │   Device ID:          2292ae5236790b47884e37cf162dcf23bfcd1c60
│ │   Previous version:   65545
│ │   Update State:       Failed
│ │   Update Error:       failed to run update on reboot
│ │   Last modified:      2022-10-23 09:59
│ │   GUID:               2f45f824-5964-4cf7-a20e-6b2a1e3e69f0
│ │   Device Flags:       • Internal device
│ │                       • Updatable
│ │                       • System requires external power source
│ │                       • Needs a reboot after installation
│ │                       • Device is usable for the duration of the update
│ │ 
│ └─  New version:      0.1.09
│       License:          Unknown
│       Description:      
│       The vendor did not supply any release notes.
│     
├─Intel Management Engine:
│ │   Device ID:          349bb341230b1a86e5effe7dfe4337e1590227bd
│ │   Previous version:   240.41.2142
│ │   Update State:       Success
│ │   Last modified:      2022-10-23 20:04
│ │   GUID:               666705ce-0624-4943-8738-09829a26c8ee
│ │   Device Flags:       • Internal device
│ │                       • Updatable
│ │                       • System requires external power source
│ │                       • Needs a reboot after installation
│ │                       • Device is usable for the duration of the update
│ │ 
│ └─  New version:      240.41.2142
│       License:          Unknown
│       Description:      
│       The vendor did not supply any release notes.
│     
└─System Firmware:
  │   Device ID:          a45df35ac0e948ee180fe216a5f703f32dda163f
  │   Previous version:   0.1.22
  │   Update State:       Success
  │   Last modified:      2022-10-23 20:55
  │   GUID:               a1392d82-62d5-4e24-863a-0f682993408f
  │   Device Flags:       • Internal device
  │                       • Updatable
  │                       • System requires external power source
  │                       • Needs a reboot after installation
  │                       • Cryptographic hash verification is available
  │                       • Device is usable for the duration of the update
  │ 
  └─  New version:      0.1.22
        License:          Unknown
        Description:      
        The vendor did not supply any release notes.
      
```
**update**
Just to highlight: The `get-history` returns error for the id `2292ae5236790b47884e37cf162dcf23bfcd1c60`. I am not sure if this error appeared after this 1.22 => 1.23 attempts, or if it was already there.

**Comment by hughsie** [comment]
> I used this command to force the reinstall: fwupdmgr --allow-reinstall --force

Why did you use `--force` out of interest?

**Comment by RayyanRiaz** [comment]
@hughsie thanks for the quick comment :)
First I tried it without `--force` flag, and then with `--force` flag because I thought that if a partial update is applied, there might be some runtime check that is blocking the update.

**Comment by mrhpearson** [comment]
Sorry - I keep finding issues that I've missed :(

@kmauleon or @ChiWei-Chen - is this one you can help with? I know we've had a few ME update issues on other platforms but wasn't sure if P17 G2 was one of them.

Mark

**Comment by kmauleon** [comment]
I think we have yet again another set of BIOS issue?  this could be the same as #250 #303 #304 in different platforms

**Comment by kmauleon** [comment]
can you please try as suggested here >> https://github.com/fwupd/firmware-lenovo/issues/295#issuecomment-1362651135

**Comment by RayyanRiaz** [comment]
Just tried that, although I have tried that already as mentioned in the issue description. Also added the line `options snd-hda-intel model=auto`  as mentioned [here](https://github.com/fwupd/firmware-lenovo/issues/295#issuecomment-1362651135). Unfortunately, it didn't work.

**Comment by hughsie** [comment]
IIRC, the ME controls the protected audio stream -- so it's not unexpected the audio is broken for a ME failed update.

**Comment by kmauleon** [comment]
another suggestion for those still has issues >> https://github.com/fwupd/firmware-lenovo/issues/295#issuecomment-1362651135  please try... thank you very much

**Comment by RayyanRiaz** [comment]
**TLDR**: The new updates partially work

**Description**
After Lenovo Premier Support suggested,  I tried the new version of the firmware(1.24) and ME(240.42.2235). I tried that with `fwupdmgr` first, but it didn't work. Then I made a Windows 10 Go-To USB and installed the `exe` files, which has now fixed these issues. The error is gone and now everything seems to work fine :) 
(Although now there is another issue that ECP does not update from 0.1.9 to 0.1.10, but that does not seem to be directly related to the current issue)

I installed the updates in this order via Windows:
* Firmware (1.24)
* ME(240.42.2235)

Thanks to all the suggestions, and I think this issue can be closed now :)
